src/asynqp/routing.py

```diff
@@ -33,22 +37,27 @@ class Sender(object):
     def __init__(self, channel_id, protocol):
         self.channel_id = channel_id
         self.protocol = protocol
+        self.connection_exc = None
 
     def send_method(self, method):
+        if self.connection_exc is not None:
+            raise self.connection_exc
         self.protocol.send_method(self.channel_id, method)
 
+    def killall(self, exc):
+        """ Connection/Channel was closed. All subsequent requests should
+            raise an error
+        """
+        self._connection_exc = exc
```

Comment:
The usecase I want to cover is like so:
```
exchange = yield from create_exchange()
while True:
    try:
        event = yield from wait_for_event()
        exchange.publish(event)
    except AMQPError:
        exchange = reconnect_and_create_exchange()
```
In other words, those methods don't raise errors on disconnect:
* publish
* ack/reject